core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushDownDereferencesThroughWindow.java

```diff
@@ -138,6 +138,7 @@ public Result apply(ProjectNode projectNode, Captures captures, Context context)
                                                             oldFunction.getArguments().stream()
                                                                     .map(expression -> replaceExpression(expression, mappings))
                                                                     .collect(toImmutableList()),
+                                                            oldFunction.getOrderingScheme(), // TODO ?????????
```

Comment:
It is fine to leave the ordering scheme as is. No need to remap as it uses plain symbols.
It could be worthy to exclude the ordering symbols from dereference pushdown, like we do for Window partitioning and ordering symbols.

Reply:
I don't quite get it? Why are we excluding those?

Reply:
I think we exclude them because we need the whole value for partitioning / ordering anyway, so pushing down the subscript doesn't save us from reading abundant data. The same reasoning should apply to aggregate's internal ordering.

Reply:
Oh ok - it looks for `Symbol symbol = getBase(expression);` - this makes more sense

Reply:
Like this?
```java
        dereferences = dereferences.stream()
                .filter(expression -> {
                    Symbol symbol = getBase(expression);
                    // Exclude partitionBy, orderBy and synthesized symbols
                    return !specification.partitionBy().contains(symbol) &&
                            !specification.orderingScheme().map(OrderingScheme::orderBy).orElse(ImmutableList.of()).contains(symbol) &&
                            !windowNode.getWindowFunctions().values().stream().anyMatch(function -> function.getOrderingScheme().map(scheme -> scheme.orderBy().contains(symbol)).orElse(false)) &&
                            !windowNode.getCreatedSymbols().contains(symbol);
                })
```

the `!windowNode.getWindowFunctions().values().stream().anyMatch(function -> function.getOrderingScheme().map(scheme -> scheme.orderBy().contains(symbol)).orElse(false)) &&` line